askama_parser/src/expr.rs

```diff
@@ -72,17 +75,74 @@ pub enum Expr<'a> {
 }
 
 impl<'a> Expr<'a> {
-    pub(super) fn arguments(i: &'a str, level: Level) -> ParseResult<'a, Vec<Self>> {
+    pub(super) fn arguments(
+        i: &'a str,
+        level: Level,
+        is_jinja_macro: bool,
+    ) -> ParseResult<'a, Vec<Self>> {
         let (_, level) = level.nest(i)?;
+        let mut named_arguments = HashSet::new();
+        let start = i;
+
         preceded(
             ws(char('(')),
             cut(terminated(
-                separated_list0(char(','), ws(move |i| Self::parse(i, level))),
+                separated_list0(
+                    char(','),
+                    ws(move |i| {
+                        Self::arguments_parsing(
+                            i,
+                            level,
+                            &mut named_arguments,
+                            start,
+                            is_jinja_macro,
+                        )
+                    }),
+                ),
                 char(')'),
             )),
         )(i)
     }
 
+    fn arguments_parsing(
```

Comment:
I'd prefer to inline this. I wonder if we can move the `if` inside the `alt` somehow?

```rust
alt((
    match is_jinja_macro {
        true => move |i| Self::named_argument(i, level, named_arguments, start),
        false => move |i| nom::err::Error::something(), // TODO?
    },
    move |i| Self::parse(i, level),
))(i)
```

Reply:
We can actually do even better and move the `is_macro` check directly into `named_argument` to make it return an error if it's not the case. Much simpler than currently.

Reply:
I still think we should inline `arguments_parsing()`.

Reply:
I find the code a bit less nice to follow but if you prefer it inlined, let's do it! :)